### DISCUSSION TITLE
Contact convergence problem

### DISCUSSION THREAD

**Discussion by selarem** [idaholab/moose#23655] [Q&A Modules: Solid mechanics] [answered]
Hi all,
I am making an indentation test. The first time increments move the indentor vertically to touch the solid below.
I am getting :+1: 
**Linear solve did not converge due to DIVERGED_PC_FAILED iterations 0
                 PC failed due to FACTOR_NUMERIC_ZEROPIVOT 
Nonlinear solve did not converge due to DIVERGED_FNORM_NAN iterations 0
 Solve Did NOT Converge!
  Finished Solving                                                                       [  5.73 s] [   39 MB]
Aborting as solve did not converge**

while there's no contact, nothing to compute: no stress and no deformation yet !

thanks for any help,
regards,

**Comment by GiudGiud** [comment, marked as answer]
Hello

Typically this means some variables/physics are not well block restricted and you have 0s in the matrix in the linear problem (within each nonlinear iteration)

You can check for conditioning/singular values issues using the SVD preconditioner
See this page
https://mooseframework.inl.gov/releases/moose/v1.0.0/application_usage/failed_solves.html#dc5ac4cb-6b57-432f-a734-51430308cea2

Guillaume

**Reply by selarem** [reply]
I have checked this before asking. 
It is not easy to do something a little different from the tests, neither on contact nor in fracture.

**Reply by GiudGiud** [reply]
What were the results?

Another option to go around FACTOR_NUMERIC_ZEROPIVOT
is to add a nonzero quantitiy to the diagonal, using the -pc_factor_shift NONZERO petsc options/values

**Reply by selarem** [reply]
Could not make it run. attached my files if anyone could help.
Thanks
https://drive.google.com/drive/folders/1RY-UxOyC2BmHYEfsTrJk8nRkQuXgUQjM?usp=sharing

**Reply by GiudGiud** [reply]
@jiangwen84 related to the other thread

**Reply by selarem** [reply]
what other thread ? this one contains contact and I think it is because of that that we do have this problem
